lib/sles4sap/ipaddr2.pm

```diff
@@ -1915,4 +1946,244 @@ sub ipaddr2_test_other_vm {
+    my $ibs_mirror_resource_group = get_required_var('IBSM_RG');
+
+    # remove the older peering
+    my $vnet_name = qesap_az_get_vnet($ibs_mirror_resource_group);
+    qesap_az_clean_old_peerings(rg => $ibs_mirror_resource_group, vnet => $vnet_name);
+
+    my $rg = ipaddr2_azure_resource_group();
+    qesap_az_vnet_peering(source_group => $rg, target_group => $ibs_mirror_resource_group);
+}
+
+=head2 ipaddr2_add_server_repos_to_hosts
+
+    ipaddr2_add_server_repos_to_hosts();
+
+Add download.suse.de server to hosts by specifying IBSM address
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_add_server_repos_to_hosts {
+    my $ibsm_ip = get_required_var('IBSM_IP');
+    my $bastion_pubip = ipaddr2_bastion_pubip();
+    foreach my $id (1 .. 2) {
+        ipaddr2_ssh_internal(id => $id,
+            cmd => "echo \"$ibsm_ip download.suse.de\" | sudo tee -a /etc/hosts",
+            bastion_ip => $bastion_pubip);
+    }
+
+    # Add repos
+    my $count = 0;
+    my @repos = split(/,/, get_var('INCIDENT_REPO', ''));
+    while (defined(my $maintrepo = shift @repos)) {
+        next if $maintrepo =~ /^\s*$/;
+        if ($maintrepo =~ /Development-Tools/ or $maintrepo =~ /Desktop-Applications/) {
+            record_info("MISSING REPOS", "There are repos in this incident, that are not uploaded to IBSM. ($maintrepo). Later errors, if they occur, may be due to these.");
+            next;
+        }
+        my $zypper_cmd = "sudo zypper --no-gpg-checks ar -f -n TEST_$count";
+        $zypper_cmd = $zypper_cmd . " --priority " . get_var('REPO_PRIORITY') if get_var('REPO_PRIORITY');
+        $zypper_cmd = $zypper_cmd . " $maintrepo TEST_$count";
+
+        foreach my $id (1 .. 2) {
+            ipaddr2_ssh_internal(id => $id,
+                cmd => $zypper_cmd,
+                bastion_ip => $bastion_pubip);
+        }
+        $count++;
+    }
+}
+
+=head2 ipaddr2_patch_system
+
+    ipaddr2_patch_system();
+
+Patch system
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_patch_system {
+    my $bastion_pubip = ipaddr2_bastion_pubip();
+
+    my @vms = ();
+    foreach my $id (1 .. 2) {
+        my $vm_private_ip = ipaddr2_get_internal_vm_private_ip(id => $id);
+        push @vms, $vm_private_ip;
+
+        ipaddr2_ssh_internal(id => $id,
+            cmd => "sudo zypper -n ref",
+            bastion_ip => $bastion_pubip,
+            timeout => 1500);
+    }
+
+    # zypper patch
+    my $host_ip = ipaddr2_bastion_ssh_addr(bastion_ip => $bastion_pubip);
+    foreach my $vm_ip (@vms) {
+        my $remote = "-J $host_ip cloudadmin@" . "$vm_ip";
+        ssh_fully_patch_system($remote);
+    }
+
+    foreach my $vm_id (1 .. 2) {
+        # To avoid the zypper lock issue
+        script_run(
+            ipaddr2_ssh_internal_cmd(
+                id => $vm_id,
+                bastion_ip => $bastion_pubip,
+                cmd => "sudo systemctl mask packagekit; sudo systemctl stop packagekit; while pgrep packagekitd; do sleep 1; done"),
```

Comment:
Do we need it? Are other sles4sap test using it? We have it in mr_test and patterns but I'm not sure we need it here.

Reply:
yes, we need it here. Sometimes VR failed in zypper lock issue